src/sentry/workflow_engine/processors/data_condition_group.py

```diff
@@ -0,0 +1,82 @@
+from typing import Any
+
+from sentry.utils.function_cache import cache_func_for_models
+from sentry.workflow_engine.models import DataCondition, DataConditionGroup
+from sentry.workflow_engine.types import ProcessedDataConditionResult
+
+
+@cache_func_for_models(
+    [(DataConditionGroup, lambda group: (group.id,))],
+)
+def get_data_condition_group(data_condition_group_id: int) -> DataConditionGroup | None:
+    try:
+        group = DataConditionGroup.objects.get(id=data_condition_group_id)
+    except DataConditionGroup.DoesNotExist:
+        group = None
+    return group
+
+
+@cache_func_for_models(
+    [(DataCondition, lambda condition: (condition.condition_group_id,))],
+)
+def get_data_conditions_for_group(data_condition_group_id: int) -> list[DataCondition]:
+    return list(DataCondition.objects.filter(condition_group_id=data_condition_group_id))
+
+
+def process_data_condition_group(
+    data_condition_group_id: int, value
+) -> ProcessedDataConditionResult:
+    group = get_data_condition_group(data_condition_group_id)
+
+    if group is None:
+        return False, []
+
+    conditions = get_data_conditions_for_group(data_condition_group_id)
+    return evaluate_condition_group(group, value, conditions=conditions)
+
+
+def evaluate_condition_group(
+    data_condition_group: DataConditionGroup,
+    value: Any,
+    **kwargs,
+) -> ProcessedDataConditionResult:
+    """
+    Evaluate the conditions for a given group and value.
+    """
+    results = []
+    conditions = kwargs.get("conditions", None)
+
+    if conditions is None:
+        conditions = get_data_conditions_for_group(data_condition_group.id)
```

Comment:
Should we just do this by default and not bother fetching them externally? This structure allows someone to pass in conditions that don't belong to the group, which seems like a potential bug waiting to happen

Reply:
yeah - agree about the footgun in code. i'll have this always fetch the conditions - not too big of a deal with the cache.